src/pip/_internal/req/req_set.py

```diff
@@ -65,7 +66,7 @@ def add_requirement(self, install_req, parent_req_name=None,
         # environment markers.
         if install_req.link and install_req.link.is_wheel:
             wheel = Wheel(install_req.link.filename)
-            if not wheel.supported():
+            if self.check_supported_wheels and not wheel.supported():
```

Comment:
`self.check_supported_wheels` can move to the outer conditional -- saves creating a Wheel instance. :)

Reply:
The `check_supported_wheels` boolean is the gate to whether or not we raise the incompatibility exception (based on whether `--target` was used during `install`). So I don't think it can move.